File: PYTHON PROGRAMS/SortingBenchmarks .py
import random

def main():    
    integers = [random.randint(0, 100) for i in range(20)]
    print("Benchmark Array:", *integers)

    bubbleArray = list(integers)
    bubbleSwap = bubbleSort(bubbleArray)
    selectionArray = list(integers)
    selectionSwap = selectionSort(selectionArray)
    insertionArray = list(integers)
    insertionSwap = insertionSort(insertionArray)

    print("Bubble Sort swap count: " + str(bubbleSwap))
    print("Selection Sort swap count: " + str(selectionSwap))
    print("Insertion Sort swap count: " + str(insertionSwap))

def bubbleSort(bubbleArray):
    swapCount = 0
    maxElement = len(bubbleArray)-1
    while maxElement >= 0:
        index = 0
        while index <= maxElement-1:
            if bubbleArray[index] > bubbleArray[index+1]:
                temp = bubbleArray[index]
                bubbleArray[index] = bubbleArray[index+1]
                bubbleArray[index+1] = temp
                swapCount += 1
            index += 1
        maxElement -= 1
    return swapCount

def selectionSort(selectionArray):
    swapCount = 0
    startScan = 0
    while startScan < len(selectionArray) - 1:
        minIndex = startScan
        minValue = selectionArray[startScan]
        index = startScan + 1
        while index < len(selectionArray):
            if selectionArray[index] < minValue:
                minValue = selectionArray[index]
                minIndex = index
            index += 1
        if minIndex != startScan:
            selectionArray[minIndex] = selectionArray[startScan]
            selectionArray[startScan] = minValue
            swapCount += 1
        startScan += 1
    return swapCount

def insertionSort(insertionArray):
    swapCount = 0
    index = 1
    while index < len(insertionArray):
        unsortedValue = insertionArray[index]
        scan = index
        while scan > 0 and insertionArray[scan-1] > unsortedValue:
            insertionArray[scan] = insertionArray[scan-1]
            scan -= 1
            swapCount += 1
        insertionArray[scan] = unsortedValue
        index += 1
    return swapCount
main()

File: JAVA PROGRAMS/SortingBenchmarks.java
import java.util.Random;

public class SortingBenchmarks {
    public static void main(String[] args) {
        Random randomNumbers = new Random();
        
        int[] integers = new int[20];
        for (int i = 0; i < 20; i++) integers[i] = randomNumbers.nextInt(100);
        System.out.print("Benchmark Array: ");
        for (int i = 0; i < 20; i++) System.out.print(integers[i] + " ");

        int[] bubbleArray = integers.clone();
        int bubbleSwap = bubbleSort(bubbleArray);
        int[] selectionArray = integers.clone();
        int selectionSwap = selectionSort(selectionArray);
        int[] insertionArray = integers.clone();
        int insertionSwap = insertionSort(insertionArray);

        System.out.println(" ");
        System.out.println("Bubble Sort swap count: " + bubbleSwap);
        System.out.println("Selection Sort swap count: " + selectionSwap);
        System.out.println("Insertion Sort swap count: " + insertionSwap);

    }
    public static int bubbleSort(int[] integers) {
        int swapCount = 0;
        int maxElement, index, temp;
        for (maxElement = integers.length -1; maxElement >= 0; maxElement--) {
            for (index = 0; index <= maxElement -1; index++) {
                if (integers[index] > integers[index +1]) {
                    temp = integers[index];
                    integers[index] = integers[index +1];
                    integers[index +1] = temp;
                    swapCount++;
                }
            }
        }
        return swapCount;
    }
    public static int selectionSort(int[] integers) {
        int swapCount = 0;
        int startScan, index, minIndex, minValue;
        for (startScan = 0; startScan < (integers.length -1); startScan++) {
            minIndex = startScan;
            minValue = integers[startScan];
            for (index = startScan +1; index < integers.length; index++) {
                if (integers[index] < minValue) {
                    minValue = integers[index];
                    minIndex = index;
                }
            }
            if (startScan != minIndex) {
                int temp = integers[startScan];
                integers[startScan] = integers[minIndex];
                integers[minIndex] = temp;
                swapCount++;
            }
        }
        return swapCount;
    }
    public static int insertionSort(int[] integers) {
        int swapCount = 0;
        int unsortedValue, scan;
        for (int index = 1; index < integers.length; index++) {
            unsortedValue = integers[index];
            scan = index;
            while (scan > 0 && integers[scan-1] > unsortedValue) {
                integers[scan] = integers[scan -1];
                scan--;
                swapCount++;
            }
        integers[scan] = unsortedValue;
        }
        return swapCount;
    }
}